Lib/email/headerregistry.py

```diff
@@ -45,8 +45,12 @@ def __init__(self, display_name='', username='', domain='', addr_spec=None):
                 raise ValueError("Invalid addr_spec; only '{}' "
                                  "could be parsed from '{}'".format(
                                     a_s, addr_spec))
-            if a_s.all_defects:
-                raise a_s.all_defects[0]
+            relevant_defects = [
+                defect for defect in a_s.all_defects
+                if not isinstance(defect, errors.NonASCIILocalPartDefect)
+            ]
+            if relevant_defects:
+                raise relevant_defects[0]
```

Comment:
I started to write that this should have a comment because the context was lacking, but when I wrote it I realized what this really means is that NonASCIILocalPartDefect is not a true defect.  Especially when you consider that the comments say it can only happen when parsing unicode, not when parsing bytes.  So instead of doing this, now that we're doing the serialization check (which is where it becomes a true defect) we should just remove the check for this defect in _header_value_parser.

We can't remove it from errors.py for backward compatibility reasons, but it should be marked as deprecated.

Reply:
I had been wondering why there wasn't a corresponding NonASCIIDomainDefect. I've removed the parser check and restored this code in Address.\_\_init\_\_() to its original form.